syntax = "proto3";
option go_package = "./grpcPb" ;
service Greeter {
  rpc FmuSimulation(FmuSimulationRequest) returns (FmuSimulationReply) {}
  rpc GetProcessStatus(GetProcessStatusRequest) returns (GetProcessStatusReply) {}
  rpc GetAllProcessNumber(GetAllProcessNumberRequest) returns (GetAllProcessNumberReply) {}
  rpc GetResult(GetResultRequest) returns (GetResultReply) {}
  rpc ProcessOperation(ProcessOperationRequest) returns (ProcessOperationReply) {}
}

message FmuSimulationRequest {
  string uuid = 1;
  string moPath = 2;
  string className = 3;
  string userName = 4;
  float startTime = 5;
  float stopTime = 6;
  string resPath = 7;
  map<string, string> params = 8;
  float outputInterval = 9;
  float tolerance=10;
}
message FmuSimulationReply {
  string log = 1;
}

message GetProcessStatusRequest{
  string uuid = 1;
}
message GetProcessStatusReply{
  string log = 1;
  int32  exception = 2;
  int32 progress = 3;
  string processStartTime = 4;
  string state = 5;
  string processRunTime = 6;
  string resPath = 7;
}

message GetAllProcessNumberRequest{}
message GetAllProcessNumberReply{
  int32 totalTasks =1;
  int32 numOfRunningProcess =2;
  int32 numOfPendingProcess =3;
}

message GetResultRequest{
    string uuid = 1;
    string variable =2;
}
message GetResultReply{
    string log = 1;
    repeated float data=2;

}

message ProcessOperationRequest{
  string uuid = 1;
  string operationName = 2;
}
message ProcessOperationReply{
  string msg =1;
}